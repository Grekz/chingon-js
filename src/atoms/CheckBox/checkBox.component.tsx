import styled from '@emotion/styled'
import * as React from 'react'

interface IInputProps {
  id: string
  checked?: boolean
  type?: string
  onChange?: (event: React.FormEvent<HTMLInputElement>) => void
}

type InputType = React.SFC<IInputProps>

interface ILabelProps {
  children: React.ReactNode
  htmlFor?: string
}

type LabelType = React.SFC<ILabelProps>

interface ICheckBoxProps extends IInputProps, ILabelProps {}

const CheckBox = ({
  checked = false,
  id,
  onChange,
  children,
}: ICheckBoxProps) => {
  const [isChecked, setChecked] = React.useState(checked)
  onChange =
    typeof onChange === 'function'
      ? onChange
      : (event: React.FormEvent<HTMLInputElement>) => {
          setChecked(!isChecked)
        }
  return (
    <CheckBoxWrapper>
      <CheckBoxInput
        id={id}
        type="checkbox"
        checked={isChecked}
        onChange={onChange}
      />
      <CheckBoxLabel htmlFor={id}>{children}</CheckBoxLabel>
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled.div``

const CheckBoxInput: InputType = styled.input``

const CheckBoxLabel: LabelType = styled.label``

export default CheckBox
