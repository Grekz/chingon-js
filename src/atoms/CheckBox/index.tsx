export { default } from './checkBox.component'
