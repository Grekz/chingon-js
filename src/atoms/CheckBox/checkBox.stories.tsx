import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import CheckBox from '.'

storiesOf('Atoms|CheckBox', module).add('As Default', () => (
  <CheckBox checked={true} id="myCheckBox">
    Checkbox button
  </CheckBox>
))
