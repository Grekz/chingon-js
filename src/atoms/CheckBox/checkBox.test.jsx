import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import CheckBox from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<CheckBox /> rendering', () => {
  it('should render CheckBox', () => {
    const wrapper = shallow(
      <CheckBox checked id="myCheckBox">
        Checkbox button
      </CheckBox>
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <CheckBox checked id="myCheckBox">
        Checkbox button
      </CheckBox>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
