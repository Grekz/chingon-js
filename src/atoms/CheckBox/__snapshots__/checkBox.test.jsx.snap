// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> rendering should match snapshot 1`] = `
<Styled(div)>
  <Styled(input)
    checked={true}
    id="myCheckBox"
    onChange={[Function]}
    type="checkbox"
  />
  <Styled(label)
    htmlFor="myCheckBox"
  >
    Checkbox button
  </Styled(label)>
</Styled(div)>
`;
